### ISSUE TITLE
SQLITE migrations

### ISSUE THREAD

**Issue by brownrl** [spotorm/spot2#36]
Having an issue with sqlite and migrations, I create the entity, get the mapper, run the migrate(),
no posts table in my db. Here is the code. The db is there and it has correct perms and traditional PDO things work. Also I am not getting any fails or message in error log. So maybe you see something, maybe it works in MySql but not in SQLITE?. At any rate this is a very promising project to help with our projects here.

$cfg = new \Spot\Config();
// Sqlite
$cfg->addConnection('sqlite', 'sqlite://db/database.sqlite' );
$spot = new \Spot\Locator($cfg);

$mapper = $spot->mapper('Entity\Post');
$mapper->migrate();

And the entity

<?php
namespace Entity;

class Post extends \Spot\Entity
{
    protected static $table = 'posts';

```
public static function fields()
{
    return [
        'id'           => ['type' => 'integer', 'primary' => true, 'autoincrement' => true],
        'title'        => ['type' => 'string', 'required' => true],
        'body'         => ['type' => 'text', 'required' => true],
        'status'       => ['type' => 'integer', 'default' => 0, 'index' => true],
        'author_id'    => ['type' => 'integer', 'required' => true],
        'date_created' => ['type' => 'datetime', 'value' => new \DateTime()]
    ];
}
```

}

**Comment by digitaldonkey** [comment]
I think it's a PHP 7.2 only issue. 
You may do 
```
if ( !is_array( $dsn ) ) {
    return $parsed;
}
```
instead of 
```
if ( !count( $dsn ) ) {
   return $parsed;
}
```
_vendor/vlucas/spot2/lib/Config.php_

http://php.net/manual/de/migration72.incompatible.php#migration72.incompatible.warn-on-non-countable-types

**Comment by digitaldonkey** [comment]
https://github.com/spotorm/spot2/issues/264

**Comment by Pnoexz** [comment]
This is happening to me on PHP 7.2.8 but using mysql. Also happens to my coworker using PHP 7.2.6. I solved this locally by changing 
```
        if ( !count( $dsn ) ) {
```
to 
```
        if ( is_array( $dsn ) && !count( $dsn ) ) {
```

This might not be the preferred solution as this might only work with full dsn as a string.

**Comment by John-Weetaker** [comment]
> Looks like a bug in how DSN string is parsed. It works if you use the alternative syntax for configuration.
> 
> ```
> $cfg->addConnection("sqlite", [
>     "path" => "database.sqlite",
>     "driver" => "pdo_sqlite"
> ]);
> ```

Very usefull. This need to be added in Readme.md

**Comment by FlipEverything** [comment]
Merged #279.
